Source/WebCore/platform/graphics/PixelBuffer.h

```diff
@@ -77,4 +79,36 @@ class PixelBuffer : public ThreadSafeRefCounted<PixelBuffer> {
     std::span<uint8_t> m_bytes;
 };
 
+// Type to use for functions that use the PixelBuffer data as source during the call, but do not store a reference to the object or modify the data.
+class PixelBufferSourceView {
+public:
+    PixelBufferSourceView() = delete;
+    PixelBufferSourceView(const PixelBuffer& pixelBuffer)
+        : PixelBufferSourceView(pixelBuffer.format(), pixelBuffer.size(), pixelBuffer.bytes())
+    {
+    }
+    static std::optional<PixelBufferSourceView> create(const PixelBufferFormat& format, const IntSize& size, std::span<const uint8_t> bytes)
```

Comment:
Is this used anywhere?

Reply:
It's the IPC constructor that validates.